pathway-analysis/02-gene_set_enrichment_analysis.Rmd

```diff
@@ -0,0 +1,243 @@
+
+We can use the `GSEA()` function to perform GSEA with any generic set of gene sets, but there are several functions for using specific, commonly used gene sets (e.g., `gseKEGG()`).
+
+```{r run_gsea}
+gsea_results <- GSEA(geneList = lfc_vector,  # ordered ranked gene list
+                     nPerm = 1000,  # number of permutations
+                     minGSSize = 25,  # minimum gene set size
+                     maxGSSize = 500,  # maximum gene set set
+                     pvalueCutoff = 0.05,
+                     pAdjustMethod = "BH",  # Benjamini-Hochberg correction
+                     TERM2GENE = dplyr::select(hs_hallmark_df,
+                                               gs_name,
+                                               entrez_gene))
+```
+
+Let's take a look at the results.
+
+```{r view_gsea, live = TRUE, eval = FALSE}
+View(gsea_results@result %>%
+       dplyr::arrange(dplyr::desc(NES))
+)
+```
+
+Normalized enrichment scores (NES) are enrichment scores that are scaled to make gene sets that contain different number of genes comparable.
+
+### Visualizing GSEA results
+
+We can visualize GSEA results for individual pathways or gene sets using `enrichplot::gseaplot()`.
+Let's take a look at 3 different pathways -- one with a highly positive NES, one with a highly negative NES, and one somewhere in the middle -- to get more insight into how ES are calculated.
+
+#### Highly Positive NES
+
+The gene set `HALLMARK_MYC_TARGETS_V2` had high positive log2 fold changes.
+Recall a positive log2 fold change means a it had a higher expression value in _MYCN_ amplified cell lines.
+
+```{r myc_v2}
+enrichplot::gseaplot(gsea_results,
+                     geneSetID = "HALLMARK_MYC_TARGETS_V2",
+                     title = "HALLMARK_MYC_TARGETS_V2")
+```
+
+Notice how the genes that are in the gene set, indicated by the black bars, tend to be on the left side of the graph indicating that they have positive gene-level scores.
+
+#### Highly Negative NES
+
+The gene set `HALLMARK_INFLAMMATORY_RESPONSE` had a highly negative NES.
+
+```{r inflammatory}
+enrichplot::gseaplot(gsea_results,
+                     geneSetID = "HALLMARK_INFLAMMATORY_RESPONSE",
+                     title = "HALLMARK_INFLAMMATORY_RESPONSE")
+```
+
+This gene set shows the opposite pattern -- genes in the pathway tend to be on the right side of the graph.
+
+#### Somewhere in the middle
+
+A moderately negative NES is somewhere in the middle in this particular experiment.
+Let's look at `HALLMARK_P53_PATHWAY`.
+
+```{r p53}
+enrichplot::gseaplot(gsea_results,
+                     geneSetID = "HALLMARK_P53_PATHWAY",
+                     title = "HALLMARK_P53_PATHWAY")
+```
+
+Genes in the pathway are distributed more evenly throughout the ranked list, resulting in a more "middling" score.
+
+*Note: The plots returned by `enrichplot::gseaplot` are ggplots, so we can use `ggplot2::ggsave()` to save them to file.*
```

Comment:
We could also probably make them less ugly! The green, it burns!

Reply:
The lines are now blue 🔵